### PR TITLE
Esri datasets, continued

### LINKED ISSUE #153: Reword 'Remove this feature' to better track with what it actually does.
We don't actually use the 'remove' button to train our AI model. This is TBD. in the meantime we should just alter the wording.

### LINKED ISSUE #157: Add 'preview mode' to Esri/Rapid integration
We need some method of enabling a 'preview mode' build so that the Esri folks can browse through preview data sets and QA them before launch.

### LINKED ISSUE #166: Back-port the Esri MVP changes into the v2 branch.
Given our decision to halt v3-based work, we are going to re-release the Esri functionality as part of the v2 tracking branch. 

This means another PR with the same functionality being collided into our track_id_v.2.18.x branch!

### PR DESCRIPTION
**Pull request by bhousel**
This is the work from #129 reimplemented onto the v2.18 branch with some significant UI and code tweaks intended to hopefully make future merges go more smoothly.

closes #166 

highlights include:
* Use more consistent naming of RapiD code and translation strings. All RapiD code should be named like `rapid_*`, avoid inconsistent use of AI/ML/FB and other prefixes.
* Use more consistent dark "RapiD" style in UI elements.  Includes rounded magenta buttons similar to those used on mapwith.ai website.  This makes it more clear which features of the app are the RapiD ones, and also helps us avoid unintended style changes when upstream iD css gets changed.
* Reimplemented the "Power Users" dialog from the v3 prototype.  Beta Esri datasets are available to power users who opt in with a feature flag. (closes #157)
* "Remove This Feature" changed to "Ignore This Feature"  (closes #153)
* I rebuilt the imagery source file so we get the updated Mapbox locator overlay.

I will continue to tweak this, but it's ready for wider testing..  (For now, poweruser features are enabled).

Screen Shot 2020-10-13 at 12 28 31 PM

Screen Shot 2020-10-13 at 12 44 28 PM

Screen Shot 2020-10-13 at 12 44 49 PM

### CODE REVIEW

File: modules/services/fb_ai_features.js [hunk 1]
```diff
@@ -8,37 +8,80 @@ import { osmEntity, osmNode, osmWay } from '../osm';
...
-        result += '&allow_tags[]=' + tag;
-      });
+    // var fb_ml_road_url = utilStringQs(window.location.hash).fb_ml_road_url;
+
+    qs.bbox = extent.toParam();
+
+    // var result = (fb_ml_road_url ? fb_ml_road_url : API_URL) + '&bbox=' + ;
+    if (taskExtent) qs.crop_bbox = taskExtent.toParam();
+
+    // TODO: FIX
```
Comment (Bonkles) [inline modules/services/fb_ai_features.js:60]: Should we chat about this TODO? I'm not sure this even gets used anymore. Probably another candidate for internal/power user features.

File: modules/services/fb_ai_features.js [hunk 2]
```diff
@@ -64,7 +107,11 @@ function getTags(obj) {
     var tags = {};
     for (var i = 0, l = elems.length; i < l; i++) {
         var attrs = elems[i].attributes;
-        tags[attrs.k.value] = attrs.v.value;
+        var k = (attrs.k.value || '').trim();
```
Comment (Bonkles) [inline modules/services/fb_ai_features.js:110]: Totally not a big deal, but were there instances of tag keys/values actually containing trailing whitespace? I'd have thought trailing whitespace wouldn't be transmitted over the wire.

File: modules/services/fb_ai_features.js [hunk 3]
```diff
@@ -173,56 +232,66 @@ export default {
...
+        Object.values(_datasets).forEach(function(ds) {
+            if (ds.cache.inflight) {
+                Object.values(ds.cache.inflight).forEach(abortRequest);
+            }
+            ds.graph = coreGraph();
+            ds.tree = coreTree(ds.graph);
+            ds.cache = { inflight: {}, loaded: {}, seen: {}, origIdTile: {} };
+        });
+
+//todo: fix
```
Comment (Bonkles) [inline modules/services/fb_ai_features.js:245]: This TODO as well. anything need to be done here?

File: modules/ui/rapid_feature_toggle_dialog.js [hunk 4]
```diff
@@ -1,192 +1,323 @@
...
-            handleToggleAllClick();
-        }
-    }
+  return function render(selection) {
+    _modalSelection = uiModal(selection);
 
+    _modalSelection.select('.modal')
+      .attr('class', 'modal rapid-modal');   // RapiD styling
 
-    function redrawOnToggle() {
```
Comment (Bonkles) [inline modules/ui/rapid_feature_toggle_dialog.js:42]: I particularly like the addition of the enter/exit() clauses for each separate layer, just so we don't have to come up with `n` different css 'hide' classes for each layer the user wants to see!

File: modules/ui/rapid_poweruser_features_dialog.js [hunk 5]
```diff
@@ -0,0 +1,212 @@
...
+  //     modal: modal,
+  //     id: 'rapid-poweruser-features-tag-sources',
+  //     label: t('rapid_poweruser_features.tag_sources'),
+  //     description: t('rapid_poweruser_features.tag_sources_desc'),
+  //     handler: toggleTagSources,
+  //     enabled: storage.featureEnabled('tagSources'),
+  //     greyout: false,
+  //     imgid: 'tag-sources',
+  //   });
+  // };
```
[the remark this answers is not in the document]
Comment (Bonkles) [reply modules/ui/rapid_poweruser_features_dialog.js:210]: Also, it looks like we got rid of one of the power user features- did QA not need one of them anymore?

**Review by Bonkles — CHANGES_REQUESTED** [review]
This is amazeballs. Props. Kudos. Many wonderful things in here, nothing that is a blocker to merging IMO. 

There are a few TODO's that are still commented, and I had a couple minor FYI-style questions here & there. 

Seriously nice work.

**Comment by Bonkles** [comment]
Is there anything remaining to be done here? I think we just need a conflict resolution and then we can merge with main, then turn the release crank.

**Comment by bhousel** [comment]
> Is there anything remaining to be done here? I think we just need a conflict resolution and then we can merge with main

I'll cleanup the commented code, restore the missing poweruser feature and merge it today.

File: modules/services/fb_ai_features.js (hunk 3 above)
Comment (bhousel) [reply modules/services/fb_ai_features.js:245]: As far as I can tell, the checkpoint stuff was there for the RapiD chapter of the walkthrough, but we don't need it now.  I setup the walkthrough graph as a separate dataset.  Now that we routinely juggle multiple datasets, adding another one for the walkthrough data is a piece of cake.

File: modules/services/fb_ai_features.js (hunk 1 above)
Comment (bhousel) [reply modules/services/fb_ai_features.js:60]: I'll leave this one commented for now - I agree it's probably not used.

**Pull request merged**